k2/csrc/array.h

```diff
@@ -178,7 +178,7 @@ class Array1 {
       K2_CHECK(new_size >= 0);
     } else {
       size_t cur_bytes_used = byte_offset_ + sizeof(T) * (size_t)dim_,
```

Comment:
std::size_t

the `size_t` in `(size_t)dim_` is redundant and can be removed.

Reply:
Seems it's code in Dan's last PR and I forgot to touch it. Fixed, thanks.